test/Tests/src/Data/Text_Spec.enso

```diff
@@ -57,8 +57,11 @@ type Manual
      - Note that currently the regex-based operations may not handle the edge
        cases described above too well.
 spec =
-    check_span result span = result.span 0 . to_grapheme_span . should_equal span
-    check_span_all result spans = result . map (m-> (m.span 0).to_grapheme_span) . should_equal spans
+    check_span match span =
+        match . should_not_equal Nothing
+        match.span 0 . should_equal span
+    check_span_all match spans = match . map (m-> (m.span 0)) . should_equal spans
```

Comment:
Does TRegex return matches indexed by Unicode Grapheme Clusters??

My hunch is that it doesn't, and that it returns UTF-16 code unit indices, like Java Regex instead.

That is why the `Utf_16_Span.to_grapheme_span` conversion had to be used here and that's why the matching functions were returning `Utf_16_Span` instances. If the new implementation wants to return `Span`s (seems like an OK thing to do), it needs to do this conversion too.

This was partially tested by the accent matching tests. Given that we disable these due to normalization problems, I'm afraid we do not have anything that tests these.

So please add some test cases which check a scenario like this one:

```
text = 'e\u0301a\u0301xyzw'
match = text.find 'yz'
text.take match.range . should_equal 'yz'
match.range . should_equal (3.up_to 5)
```

This verifies that `e\u0301` which is **2 UTF-16 code units** is correctly counted as **1 grapheme cluster** and double-checks that the range returned is consistent with our other methods, like `Text.take` - which are all grapheme cluster based.

If the conversion is not done properly, I'd expect the span to come out as `5-7` (counting code units) and so `text.take match.range` would wrongly return `w`.

Reply:
Ideally also please add a test like:

```
text2 = 'a\u0301e\u0301bcde\u0301fghijklm'
match2 = text2.find "d.*h" . span 0
match2.range . should_equal (4.up_to 9)
match2.length . should_equal 5
text2.take match2.range . should_equal "défgh"
```

to verify that the length of the span is also correctly counted if it contains multi-codepoint grapheme clusters inside it.

Reply:
:) While you were typing this I was deep in figuring this out and adding tests, please take a look.

Reply:
Looks great!